paper-server/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java

```diff
@@ -3541,4 +3543,35 @@ public void sendEntityEffect(final org.bukkit.EntityEffect effect, final org.buk
         this.getHandle().connection.send(new net.minecraft.network.protocol.game.ClientboundEntityEventPacket(((CraftEntity) target).getHandle(), effect.getData()));
     }
     // Paper end - entity effect API
+
+
+    @Override
+    public @NotNull List<@NotNull Item> give(final @NotNull ItemStack @NotNull ... items) {
+        return this.give(List.of(items), true);
+    }
+
+    @Override
+    public @NotNull List<@NotNull Item> give(@NotNull final Collection<@NotNull ItemStack> items) {
+        return this.give(items, true);
+    }
+
+    @Override
+    public @NotNull List<@NotNull Item> give(@NotNull final Collection<@NotNull ItemStack> items, final boolean dropIfFull) {
+        Preconditions.checkNotNull(items, "items cannot be null");
+        ImmutableList.Builder<Item> droppedItems = ImmutableList.builder();
+        for (ItemStack item : items) {
+            Preconditions.checkNotNull(item, "ItemStack cannot be null");
+            Preconditions.checkArgument(!item.isEmpty(), "ItemStack cannot be empty");
+            Preconditions.checkArgument(item.getAmount() <= item.getMaxStackSize(), "ItemStack amount cannot be greater than its max stack size");
```

Comment:
Why? or this mess with the add logic?
if so then the docs can add the throws for this cases.